core/src/main/java/jenkins/model/TopLevelItemLoader.java

```diff
@@ -0,0 +1,12 @@
+package jenkins.model;
+
+import hudson.ExtensionPoint;
+import hudson.model.TopLevelItem;
+
+import java.io.IOException;
+import java.util.Collection;
+
+
+public interface TopLevelItemLoader extends ExtensionPoint {
```

Comment:
Needs documentation.

It's also a good practice to assign <code>ExtensionPoint</code> to classes instead of interfaces. Otherwise you won't be able to add new methods without breaking the binary compatibility.